getting_started/scripting/gdscript/gdscript_styleguide.rst

```diff
@@ -194,3 +207,36 @@ Constants
 
 Use CONSTANT\_CASE, all caps, with an underscore (\_) to separate words:
 ``const MAX_SPEED = 200``
+
+Static Typing
+-------------
+
+Since Godot 3.1, GDScript supports :ref:`optional static typing<_doc_gdscript_static_typing>`.
```

Comment:
The style guide doesn't specify whether optional type hints should be used in the documentation when writing tutorials or code examples in the class reference. This is maybe not the right place for it though as this is a general style guide for engine users, but making a note here to make sure that we have some guidelines about this.

Reply:
Agreed, this is outside the scope of the style guide, which exists to prescribe how GDScript should look, whether using optional type hints or not.

The question of whether type hints belong in tutorials and code snippets throughout the docs is another conversation entirely. For what it's worth, I'm against that, but we can debate that in a separate issue.

Reply:
I agree with Chris on this one, I wouldn't have type hints in the docs. Dynamic gdscript is easy on the eye, easy to read, so it helps to keep the manual short. Type hints are an optional feature, the docs should reflect that. Switching at this point would be a lot of work and I would expect it to make beginners more likely to quit reading the docs.